### PR TITLE
[tensorflow][build] Expose tunable parameters to support multiple tfs

### PR DESCRIPTION
**Pull request by jinpengqi**
*Issue #, if available:*

## PR Checklist

- [x]  I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]

- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Checklist
* When creating a PR:
- [ ] I've modified `src/config/build_config.py` in my PR branch by setting `ENABLE_EI_MODE = True` or `ENABLE_NEURON_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above
#### Benchmark Checklist
* When creating a PR:
- [ ] I've modified `src/config/test_config.py` in my PR branch by setting `ENABLE_BENCHMARK_DEV_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*
Syncing with the pr in sagemaker-tensorflow-serving-container: [link](https://github.com/aws/sagemaker-tensorflow-serving-container/pull/190/commits/a936f110f7b0805c0b538b0a3c1db15415424e75)

Implement new container environment which could expose additional parameters to help user optimize for throughput and fine tune container performance.

The feature that multiple of tensorflow model servers could handle parallel requests coming from different gunicorn workers could be supported in this release. And throughput will be improved based on different parameters tuned.
Specifically, tunable parameters likeSAGEMAKER_GUNICORN_WORKERS, SAGEMAKER_TFS_INSTANCE_COUNT, SAGEMAKER_TFS_INTER_OP_PARALLELISM, SAGEMAKER_TFS_INTRA_OP_PARALLELISM and etc.

*Tests run:*
* build images locally and run inference against it succeed 
* Tested on sageMaker endpoint and observed no failure & improved inference performance.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tensorflow/inference/docker/build_artifacts/sagemaker/serve.py [hunk 1]
```diff
@@ -137,22 +152,34 @@ def _setup_gunicorn(self):
                         raise ChildProcessError('failed to install required packages.')
 
         gunicorn_command = (
-            'gunicorn -b unix:/tmp/gunicorn.sock -k gevent --chdir /sagemaker '
-            '{}{} -e TFS_GRPC_PORT={} -e SAGEMAKER_MULTI_MODEL={} -e SAGEMAKER_SAFE_PORT_RANGE={} '
-            'python_service:app').format(python_path_option, ','.join(python_path_content),
-                                         self._tfs_grpc_port, self._tfs_enable_multi_model_endpoint,
+            "gunicorn -b unix:/tmp/gunicorn.sock -k {} --chdir /sagemaker "
+            "{}{} -e TFS_GRPC_PORT_RANGE={} -e TFS_REST_PORT_RANGE={} "
```
Comment (liangma8712) [inline tensorflow/inference/docker/build_artifacts/sagemaker/serve.py:156]: can we use SAGEMAKER_GUNICORN_WORKERS and SAGEMAKER_GUNICORN_THREADS to configure gunicorn here as well?

**Review by liangma8712 — CHANGES_REQUESTED** [review]
looks good overall.

File: tensorflow/inference/docker/build_artifacts/sagemaker/serve.py [hunk 2]
```diff
@@ -11,6 +11,7 @@
 # ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
 
+import boto3
```
Comment (liangma8712) [inline tensorflow/inference/docker/build_artifacts/sagemaker/serve.py:14]: I think boto3 is not used in this code?

File: tensorflow/inference/docker/build_artifacts/sagemaker/serve.py (hunk 1 above)
Comment (jinpengqi) [reply tensorflow/inference/docker/build_artifacts/sagemaker/serve.py:159]: yep, added back

File: tensorflow/inference/docker/build_artifacts/sagemaker/serve.py (hunk 2 above)
Comment (jinpengqi) [reply tensorflow/inference/docker/build_artifacts/sagemaker/serve.py]: removed

**Review by liangma8712 — DISMISSED** [review]

**Review by liangma8712 — APPROVED** [review]

**Review by saimidu — APPROVED** [review]

**Pull request merged**